src/web-components/SelectHandle/index.ts

```diff
@@ -0,0 +1,191 @@
+    validUntilDate?: number;
+}
+
+interface BasicAsset {
+    policyId: string;
+    hex: string;
+}
+
+/**
+ * `select-handle` is a custom LitElement component for selecting wallet handles.
+ * 
+ * ### Slots:
+ * - **`slottedSearch`**: Slot for the search input.
+ * - **`slottedButtons`**: Slot for action buttons.
+ * - **`slottedLoader`**: Slot for a loading spinner or indicator.
+ * 
+ * ### Properties:
+ * - **`handleData`**:  
+ *   The data for the handle, expected to follow the `GetHandleResponse` format.
+ * - **`addFunction`**:  
+ *   A function called during the `firstUpdated()` lifecycle method.
+ * - **`infiniteScroll`**:  
+ *   A function or property that handles infinite scrolling behavior.
+ * - **`slottedButtonsStyling`**:  
+ *   A string used to apply custom CSS styling to the slotted buttons.
+ * - **`slottedSearchStyling`**:  
+ *   A string used to style the slotted search input.
+ * - **`selectHandle`**:  
+ *   A function that processes the selected handle. It receives the handle object and sends it back to the parent component.
+ * - **`loadingImg`**:  
+ *   A boolean flag indicating whether to display a loading spinner. The loading spinner content must be provided through the `slottedLoader` slot.
+ * 
+ * ### Example Method:
+ * ```js
+ * function selectHandle(handle) {
+ *   this.loadingImg = true;
+ *   console.log('Selected handle:', handle);
+ *   this.loadingImg = false;
+ * }
+ * ```
+ * 
+ * ### Example Usage:
+ * ```html
+ * <select-handle
+ *   .handleData=${handleData}
+ *   .addFunction=${addFunction}
+ *   .infiniteScroll=${infiniteScroll}
+ *   .selectHandle=${selectHandle}
+ *   .slottedButtonsStyling=${'display: flex;'}>
+ * </select-handle>
+ * ```
+ */
+
+
+@customElement('select-handle')
+export class SelectHandle extends LitElement {
+    @property({ type: Array }) handleData: any[] = [];
+    @property({ type: Object }) handle: any = {};
+    @property({ type: Boolean }) dropdownOpen = false;
+    @property({ type: Boolean }) loadingImg = false;
+    @property({ type: Boolean }) isLoading = false;
+    @property({ type: String }) route = '';
+    @property({ type: String }) slottedButtonsStyling: string;
+    @property({ type: String }) slottedSearchStyling: string;
+    @property({ type: Function }) addFunction = () => { };
+    @property({ type: Function }) infiniteScroll = () => { };
+    @property({ type: Function }) selectHandle = (handle) => { };
+    @property({ type: Object })
+    litElement!: LitElement;
+    help: string;
+
+    static styles = SelectHandleStyles
+
+    firstUpdated() {
+        const handle = localStorage.getItem('selectedHandle');
+        this.handle = JSON.parse(handle ?? '{}');
+        this.addFunction();
+    }
+
+    imageUrl = (img: string): string => {
+        const image = img.replace('ipfs://', '') || '';
+        if (!image) {
+            return '';
+        }
+        return `${IPFS_GATEWAY}/ipfs/${image}${IPFS_GATEWAY_RESIZE_QUERY}`;
+    }
```

Comment:
I think we should pass the imageUrl in instead of using the gateway here. This way, if we need to try a different gateway we can.